interfaces/cython/cantera/thermo.pyx

```diff
@@ -1564,9 +1565,33 @@ cdef class PureFluid(ThermoPhase):
             return self.T, self.density, self.X
 
     property TPX:
-        """Get the temperature [K], pressure [Pa], and vapor fraction."""
+        """
+        Get/Set the temperature [K], pressure [Pa], and vapor fraction of a
+        PureFluid.
+
+        An Exception is raised if the thermodynamic state is not consistent.
+        """
         def __get__(self):
             return self.T, self.P, self.X
+        def __set__(self, values):
+            T = values[0] if values[0] is not None else self.T
+            P = values[1] if values[1] is not None else self.P
+            X = values[2] if values[2] is not None else self.X
+            if not isinstance(X, (np.ndarray, _numbers.Number)):
+                raise ValueError(
```

Comment:
I don't think this needs to be on a new line, at least at this length. Can you change `X` to `vapor fraction (X)`? I think that would help clarify.